activitysim/abm/models/util/tour_frequency.py

```diff
@@ -435,3 +435,240 @@ def process_joint_tours(joint_tour_frequency, joint_tour_frequency_alts, point_p
+
+    return tours
+
+def create_joint_tours(tour_counts, tour_category, parent_col='person_id'):
+    """
+    This method processes the tour_frequency column that comes
+    out of the model of the same name and turns into a DataFrame that
+    represents the tours that were generated
+    
+    Parameters
+    ----------
+    tour_counts: DataFrame
+        table specifying how many tours of each type to create
+        one row per person (or parent_tour for atwork subtours)
+        one (int) column per tour_type, with number of tours to create
+    tour_category : str
+        one of 'mandatory', 'non_mandatory', 'atwork', or 'joint'
+    
+    Returns
+    -------
+    tours : pandas.DataFrame
+        An example of a tours DataFrame is supplied as a comment in the
+        source code - it has an index which is a unique tour identifier,
+        a person_id column, and a tour type column which comes from the
+        column names of the alternatives DataFrame supplied above.
+        
+        tours.tour_type       - tour type (e.g. school, work, shopping, eat)
+        tours.composition     - tour composition (e.g. adults, children, mixed)
+        tours.tour_type_num   - if there are two 'school' type tours, they will be numbered 1 and 2
+        tours.tour_type_count - number of tours of tour_type parent has (parent's max tour_type_num)
+        tours.tour_num        - index of tour (of any type) for parent
+        tours.tour_count      - number of tours of any type) for parent (parent's max tour_num)
+        tours.tour_category   - one of 'mandatory', 'non_mandatory', 'atwork', or 'joint'
+    """
+
+    # FIXME - document requirement to ensure adjacent tour_type_nums in tour_num order
+
+    """
+               alt1       alt2     alt3
+    <parent_col>
+    2588676       2         0         0
+    2588677       1         1         0
+    """
+    tour_type_dict={5:'shopping',6:'othmaint',7:'eatout',8:'social',9:'othdiscr'}
+    tour_comp_dict={1:'adults',2:'children',3:'mixed'}
+
+    # reformat with the columns given below
+    tours_purp = tour_counts[['purpose1','purpose2']].stack().reset_index()
+    tours_purp.columns = [parent_col, "tour_id_temp", "tour_type"]
+    tours_purp['tour_id_temp'] = range(1, 1+len(tours_purp))
+    tours_purp['tour_type'] = tours_purp['tour_type'].map(tour_type_dict)
+
+    """
+        <parent_col> tour_id_temp  tour_type
+    0     2588676    purpose1           5
+    1     2588676    purpose2           0
+    2     2588677    purpose1           5
+    3     2588677    purpose2           5
+    4     2588678    purpose1           6
+    5     2588678    purpose2           7
+    
+    parent_col is the index from non_mandatory_tour_frequency
+    tour_type is the column name from non_mandatory_tour_frequency_alts
+    tour_type_count is the count value of the tour's chosen alt's tour_type from alts table
+    """
+    tours_comp = tour_counts[['party1','party2']].stack().reset_index()
```

Comment:
Can we make this more generalizable incase we have more than 2 parties? Same thing above with purpose1 and purpose2.  Also, looks like these columns are getting read-in from the alternatives table.  Seems like we should include an assert statement that checks for the purpose and party columns in the alts table in that case.